### ISSUE TITLE
README.solaris

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4330]
Migrated from [rt.perl.org#7558](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7558) (status was 'resolved')

Searchable as RT7558$

**Comment by p5pRT** [comment]
### From Martin.Bellenberg@t-systems.de



---
$Id&#8203;: README\.solaris\,v 1\.4 2000/11/11 20&#8203;:29&#8203;:58 doughera Exp $
\.\.\.
=head2 Net&#8203;::SSLeay

Net&#8203;::SSLeay requires a /dev/urandom to be present\. This device is not
part of Solaris\. You can either get the package SUNWski \(packaged with
several Sun software products\, for example the Sun WebServer\, which is
part of the Solaris Server Intranet Extension\, or the Sun Directory
Services\, part of Solaris for ISPs\) or download the ANDIrand package
from L\<http&#8203;://www\.cosy\.sbg\.ac\.at/~andi/>\. If you use SUNWski\, make a
symbolic link /dev/urandom pointing to /dev/random\.


---
I would suggest to also mention\, that the package SUNWski and related ones can be easily found as part of SUN patch 105710\-01\, 106755\-01\, 106756\-01 or 106754\-01\. These patches might be fetched at
ftp&#8203;://sunsolve\.sun\.com/pub/patches or related country sites\. Packages included and needed due to dependencies are&#8203;:
application SUNWski        SKI 1\.0 Software \(User Package\)
application SUNWskica      SKI 1\.0 Software \(CA Package\)
application SUNWskicw      SKI 1\.0 Software \(Licensing Package for CA\)
application SUNWskimc      SKI 1\.0 Software \(CA Manual Page Package\)
application SUNWskimu      SKI 1\.0 Software \(User Manual Page Package\)
application SUNWssl        SSL 1\.0 Software \(Library Global Version\)

greetings
Martin Bellenberg

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

Martin Bellenberg
Diplom\-Geophysiker

T\-Systems
Service & Netze\, Unix/NT\-Systemadministration

Geschäftsstelle Hamburg
debis Systemhaus GEI GmbH
Lademannbogen 21\-23\, D\-22339 Hamburg

fon&#8203;: \+49 40 5395\-1571
fax&#8203;: \+49 40 5395\-1733

mailto&#8203;:Martin\.Bellenberg@&#8203;t\-systems\.de
http&#8203;://www\.t\-systems\.de
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

**Comment by p5pRT** [comment]
### From @doughera88

I think this patch addresses the concerns in bug perl\#7558\,
and that bug may be closed once this is applied\.

<details><summary>Inline Patch</summary>

```diff
--- perl-5.8.5/README.solaris	2003-08-25 17:06:34.000000000 -0400
+++ perl-5.8.x-andy/README.solaris	2004-07-26 15:27:40.000000000 -0400
@@ -587,7 +587,10 @@
 the Solaris Server Intranet Extension, or the Sun Directory Services,
 part of Solaris for ISPs) or download the ANDIrand package from
 L<http://www.cosy.sbg.ac.at/~andi/>. If you use SUNWski, make a
-symbolic link /dev/urandom pointing to /dev/random.
+symbolic link /dev/urandom pointing to /dev/random.  For more details,
+see Document ID27606 entitled "Differing /dev/random support requirements
+within Solaris[TM] Operating Environments", available at
+http://support.sunsoft.com.

 It may be possible to use the Entropy Gathering Daemon (written in
 Perl!), available from L<http://www.lothar.com/tech/crypto/>.

-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @iabyn

On Mon\, Jul 26\, 2004 at 03&#8203;:32&#8203;:35PM \-0400\, Andy Dougherty wrote&#8203;:
> I think this patch addresses the concerns in bug perl\#7558\,
> and that bug may be closed once this is applied\.

Thanks\, patch applied\, bug report closed\.

> 
> \-\-\- perl\-5\.8\.5/README\.solaris    2003\-08\-25 17&#8203;:06&#8203;:34\.000000000 \-0400
> \+\+\+ perl\-5\.8\.x\-andy/README\.solaris    2004\-07\-26 15&#8203;:27&#8203;:40\.000000000 \-0400
> @&#8203;@&#8203; \-587\,7 \+587\,10 @&#8203;@&#8203;
>  the Solaris Server Intranet Extension\, or the Sun Directory Services\,
>  part of Solaris for ISPs\) or download the ANDIrand package from
>  L\<http&#8203;://www\.cosy\.sbg\.ac\.at/~andi/>\. If you use SUNWski\, make a
> \-symbolic link /dev/urandom pointing to /dev/random\.
> \+symbolic link /dev/urandom pointing to /dev/random\.  For more details\,
> \+see Document ID27606 entitled "Differing /dev/random support requirements
> \+within Solaris\[TM\] Operating Environments"\, available at
> \+http&#8203;://support\.sunsoft\.com\.
> 
>  It may be possible to use the Entropy Gathering Daemon \(written in
>  Perl\!\)\, available from L\<http&#8203;://www\.lothar\.com/tech/crypto/>\.
> 
> \-\- 
>     Andy Dougherty        doughera@&#8203;lafayette\.edu

\-\- 
Never do today what you can put off till tomorrow\.

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @AlanBurlison

> On Mon\, Jul 26\, 2004 at 03&#8203;:32&#8203;:35PM \-0400\, Andy Dougherty wrote&#8203;:
> > I think this patch addresses the concerns in bug perl\#7558\,
> > and that bug may be closed once this is applied\.
> 
> Thanks\, patch applied\, bug report closed\.

Solaris 9 onwards provide /dev/u?random as standard\, and perl finds and
uses it OK&#8203;:

$ truss \-t open perl \-e rand 2>&1 | grep random
open64\("/dev/urandom"\, O\_RDONLY\)        = 3

**Comment by p5pRT** [comment]
### From @doughera88

On Tue\, 27 Jul 2004\, Bellenberg\, Martin wrote&#8203;:

> Dear Andy\,
>
> Concerning&#8203;: http&#8203;://rt\.perl\.org/rt3/index\.html?q=7558
>
> sorry\, but the patch in my opinion does not reflect my suggestion to
> mention certain solaris patches to gain Sun conform installation of
> /dev/random in Solaris 2\.5\.1\, 2\.6 and 7\.

I see your point\, but your suggestions \(naturally\) reflected the state of
affairs at the time you made them\.  It was my hope that the document I
intended to reference would stay more up\-to\-date and hence perl's
README\.solaris would need no further editing or maintenance\.

> Furthermore Sun Patch 112438\-03 \(SPARC\) or 112439\-02 \(x86\) installs
> /dev/random by one easy 'patchadd' in Solaris 8\, patches are available
> to contract customers only\.

I was reluctant to mention a specific patch id or give a specific URL
since both can get out\-of\-date\, though the document I intended to
reference does mention patch 112438\.  It also gives more up\-to\-date
information on Solaris versions beyond 7\.

> Also URL http&#8203;://support\.sunsoft\.com does not exist\.

Well\, that's obviously quite bad \-\- my mistake\!  Thanks for catching it\.

Additional patch to fix the url attached&#8203;:

<details><summary>Inline Patch</summary>

```diff
--- perl-current/README.solaris	Mon Jul 26 18:32:06 2004
+++ perl-andy/README.solaris	Tue Jul 27 14:03:26 2004
@@ -590,7 +590,7 @@
 symbolic link /dev/urandom pointing to /dev/random.  For more details,
 see Document ID27606 entitled "Differing /dev/random support requirements
 within Solaris[TM] Operating Environments", available at
-http://support.sunsoft.com.
+http://sunsolve.sun.com .

 It may be possible to use the Entropy Gathering Daemon (written in
 Perl!), available from L<http://www.lothar.com/tech/crypto/>.


-- 
```
</details>

&nbsp;    Andy Dougherty        doughera@&#8203;lafayette\.edu

**Comment by p5pRT** [comment]
### From @rgs

Andy Dougherty wrote&#8203;:
> \-\-\- perl\-current/README\.solaris    Mon Jul 26 18&#8203;:32&#8203;:06 2004
> \+\+\+ perl\-andy/README\.solaris    Tue Jul 27 14&#8203;:03&#8203;:26 2004
> @&#8203;@&#8203; \-590\,7 \+590\,7 @&#8203;@&#8203;
>  symbolic link /dev/urandom pointing to /dev/random\.  For more details\,
>  see Document ID27606 entitled "Differing /dev/random support requirements
>  within Solaris\[TM\] Operating Environments"\, available at
> \-http&#8203;://support\.sunsoft\.com\.
> \+http&#8203;://sunsolve\.sun\.com \.

Thanks\, applied as \#23165\.
